crates/sui-indexer/src/models/checkpoint_metrics.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use diesel::prelude::*;
+use diesel::sql_types::Float8;
+
+use crate::schema::checkpoint_metrics;
+
+#[derive(Queryable, Insertable, Debug, Clone)]
+#[diesel(table_name = checkpoint_metrics)]
+pub struct CheckpointMetrics {
+    pub checkpoint: i64,
+    pub epoch: i64,
+    pub real_time_tps: f64,
+    pub peak_tps_30d: f64,
+    pub rolling_total_transactions: i64,
+    pub rolling_total_transaction_blocks: i64,
+    pub rolling_total_successful_transactions: i64,
+    pub rolling_total_successful_transaction_blocks: i64,
+}
+
+impl Default for CheckpointMetrics {
+    fn default() -> Self {
+        Self {
+            // -1 to differentiate from first checkpoint of sequence 0
+            checkpoint: -1,
```

Comment:
If we want to represent nullability then we should probably use Option here vs using -1 to denote it missing

Reply:
agreed that while reading None will be more explicit, this field is not optional when the row exists, so we do not want to make the field optional.

on the reading path, the no-such-row is handled via `.unwrap_or_default()` where the default value here is an convenient accumulation seed with checkpoint & rolling* values.